src/System.IO/src/System/IO/BufferedStream.cs

```diff
@@ -234,7 +236,11 @@ protected override void Dispose(bool disposing)
             finally
             {
                 _stream = null;
-                _buffer = null;
+                if (_buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
I noted this offline, but I think that:
a) We should really change the default for clearArray to be true rather than false, so that by default code gets the more secure behavior and opts-in to trading that off for better perf, rather than the other way around.
b) And regardless of (a), we should always be clearing when returning buffers used in our libraries and when those buffers were storing any user-provided data.